### PR TITLE
feat: implement more conservative collateral requirements for long risk-partnered positions

### PR DESCRIPTION
**Pull request by dyedm1**
Spreads - Changed the "base" collateral requirement to be that of the long leg. The long leg caps the maximum risk of the short leg, but the long leg itself still has premium risk. The new collateral requirement is the long leg requirement + 100% collateralized spread risk (token delta between the bounds of the "risky" portion of the spread)

Strangles/straddles - removed long strangles from the set of valid positions. Unlike short strangles, long strangles do not have any meaningful difference in risk (especially straddles - premium is accumulated at twice the rate) because the risk is derived only from the premium and not the intrinsic value of the position. 

Thanks to Robert from Numoen for asking an important question. This PR makes the answer 40x instead of infinity :)
image

### CODE REVIEW

File: contracts/CollateralTracker.sol [hunk 1]
```diff
@@ -1789,6 +1733,13 @@ contract CollateralTracker is ERC20Minimal, Multicall {
                     : ((notional - notionalP) * contracts) / notionalP;
             }
         }
+
+        // add the collateral requirement for the long leg as a base
+        spreadRequirement += _getRequiredCollateralAtUtilization(
```
Comment (guil-lambert) [inline contracts/CollateralTracker.sol]: Do we want to make it: `spreadRequirement = max(longLegRequirement, movedPartnerRight - movedRight)`

Would make it worthwhile for a bit longer to create spreads vs. the naked option --ie. spreads up to 20% wide vs. spreads up to 10% wide assuming 10% base collateral requirement for the long option


--
Small nitpick: it is better for readability if the base requirement is added before line `1701`: 

ie. `base + extra` as opposed to `extra + base`.

**Review by guil-lambert — CHANGES_REQUESTED** [review]
See comment, everything looks good otherwise

File: contracts/CollateralTracker.sol (hunk 1 above)
Comment (dyedm1) [reply contracts/CollateralTracker.sol]: >Do we want to make it: spreadRequirement = max(longLegRequirement, movedPartnerRight - movedRight)
Would make it worthwhile for a bit longer to create spreads vs. the naked option --ie. spreads up to 20% wide vs. spreads up to 10% wide assuming 10% base collateral requirement for the long option

I'm not sure that particular formula makes sense - a 5% wide spread would have the same collateral requirement as a 10% wide while being significantly more risky. 

There's no reason why we need to force users to 100% collateralize the risky portion of the spread though when the same standard is not applied to normal positions.

What if we just made it `spreadRequirement = longLegRequirement + (movedPartnerRight - movedRight)*SCR)`? That way the collateral requirement for risk-partnered spreads is asymptotic to the non-risk-partnered requirement as the strikes approach the ends of the tickrange and there's no weird calculus about whether to use the risk partner system for your spread or not.

Comment (dyedm1) [reply contracts/CollateralTracker.sol]: >Small nitpick: it is better for readability if the base requirement is added before line 1701

Agreed

**Pull request merged**